DUI3-DX/Converters/Rhino/Speckle.Converters.Rhino7/ToHost/TopLevel/SpeckleToHostGeometryBaseConversion.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Rhino;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Kits;
+using Speckle.Core.Models;
+
+namespace Speckle.Converters.Rhino7.ToHost.TopLevel;
+
+public abstract class SpeckleToHostGeometryBaseConversion<TIn, TOut> : ISpeckleObjectToHostConversion
+  where TIn : Base
+  where TOut : RG.GeometryBase
+{
+  protected IConversionContextStack<RhinoDoc, UnitSystem> ContextStack { get; private set; }
+  private readonly IRawConversion<TIn, TOut> _geometryBaseConverter;
+
+  protected SpeckleToHostGeometryBaseConversion(
+    IConversionContextStack<RhinoDoc, UnitSystem> contextStack,
+    IRawConversion<TIn, TOut> geometryBaseConverter
+  )
+  {
+    ContextStack = contextStack;
+    _geometryBaseConverter = geometryBaseConverter;
+  }
+
+  public object Convert(Base target)
+  {
+    var castedBase = (TIn)target;
+    var result = _geometryBaseConverter.RawConvert(castedBase);
+
+    /*
+     * POC: Looking at a simpler, more performant way of doing unit scaling on `ToNative`
+     * by fully relying on the transform capabilities of the HostApp, and only transforming top-level stuff.
+     * This may not hold when adding more complex conversions, but it works for now!
+     */
+    if (castedBase["units"] is string units)
```

Comment:
Can any nesting of objects result in the units needing to be scaled based on previous scaling? Would it maybe get a bit lossy if so?

Reply:
This class is limited to output `GeometryBase` objects in Rhino, which do not have any children by default. Things like Brep (who do have inner geometry that defines the brep).

Not sure nested scaling is a thing for units 🤔 or if we want it to be. Things like blocks (which will hold geometry) will have their own transform but the geometry itself should just be scaled according to the scaling factor between it's own units and the current document ones.